scheduler/src/cook/spnego.clj

```diff
@@ -78,13 +78,14 @@
   [rh]
   (fn [req]
     (if (get-in req [:headers "authorization"])
-      (let [gss_context (gss-context-init)]
-        (if-let [token (do-gss-auth-check gss_context req)]
+      (let [gss_context (gss-context-init)
+            token (do-gss-auth-check gss_context req)]
+        (if (.isEstablished gss_context)
           (let [princ (gss-get-princ gss_context)]
-            (-> req
-                (assoc :krb5-authenticated-princ princ
-                       :authorization/user (first (str/split princ #"@" 2)))
-                (rh)
-                (header "WWW-Authenticate" token)))
+            (cond-> (-> req
+                        (assoc :krb5-authenticated-princ princ
+                               :authorization/user (first (str/split princ #"@" 2)))
+                        (rh))
+              token (header "WWW-Authenticate" token)))
```

Comment:
What is the implication of not including this WWW-Authenticate when the token doesn't exist?